File: Assets/Scripts/Runes/SetBaseMana.cs
using UnityEngine;
using System.Collections;

public class SetBaseMana : Rune {

    public string controller_uid { get; set; }
    public int base_mana { get; set; }

    public override void Execute(System.Action action)
    {
        Controller playerController = EntityManager.Singelton.GetEntity(controller_uid) as Controller;
        if(playerController == null)
        {
            Debug.Log("Program finding Controller with guid" + controller_uid);
            action();
            return;
        }
        playerController.SetBaseMana(base_mana);
        action();
    }
    public override void OnGUI()
    {
        
    }
}

File: Server/runes/SetBaseMana.js
exports.execute = function (rune, state) {
    var controller = state.controllers[rune.controllerGuid];
    controller.baseMana = rune.baseMana;
}

exports.CreateRune = function(controllerGuid, baseMana)
{
    var rune = {
        "runeType":"SetBaseMana",
        "controllerGuid":controllerGuid,
        "baseMana":baseMana
    }
    return rune;
}

exports.canSee = function (rune, controller, state) {
    return true;
}